From be87a08bc2200b80c068347568bea07a4fa09fb7 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 19 May 2017 14:39:35 +0200
Subject: [PATCH] setup-keymap: allow specify the variant together with layout

Some users may know exactly which variant they want, and instead of
asking them two questions, we accept if users types in the variant at
the first question.

if the variant does not exist, then user is prompted.
---
 setup-keymap.in | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/setup-keymap.in b/setup-keymap.in
index 67f89cb..ebcd29c 100644
--- a/setup-keymap.in
+++ b/setup-keymap.in
@@ -38,7 +38,13 @@ select_layout() {
 		default_read layout "$layout"
 		if [ "$layout" = "abort" ] || [ "$layout" = "none" ] ; then
 			goodbye 0
-		elif [ -d "$MAPDIR/$layout" ] ; then
+		fi
+		set -- $layout
+		if [ $# -eq 2 ]; then
+			layout="$1"
+			variant="$2"
+		fi
+		if [ -d "$MAPDIR/$layout" ] ; then
 			return 0
 		fi
 	done
-- 
2.13.0

